### PR TITLE
if we suspect the request to come from a bot, we'll show it a consistent...

### PR DESCRIPTION
**Pull request by eishay**
... version of the landing page (v2)

### CODE REVIEW

**Comment by eishay** [comment]
@andrewconner @joshm1 @raymondng

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala [hunk 1]
```diff
@@ -37,25 +38,37 @@ object MarketingSiteRouter extends AssetsBuilder with Controller with Logging {
...
-    } recover {
-      case t: Throwable =>
-        log.error(s"[landing] Caught exception $t while parsing queryParam(v):${request.queryString("v")}")
-        None
-    } get
+  private trait LandingVersion {
+    def version: Int
+  }
+  private object Version1 extends LandingVersion { val version = 1 }
+  private object Version2 extends LandingVersion { val version = 1 }
```
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: 2?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala [hunk 2]
```diff
@@ -37,25 +38,37 @@ object MarketingSiteRouter extends AssetsBuilder with Controller with Logging {
...
+
+  private def landing(implicit request: Request[_]): String = {
+    val possiblyBot = request.userAgentOpt.map(_.possiblyBot).getOrElse(true)
```
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala:48]: Interesting. Didn't know about `possiblyBot`.

**Comment by raymondng** [comment]
lgtm.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala (hunk 1 above)
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: good catch, this method has to be tested. i'll do it

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: +1 on using objects

**Comment by andrewconner** [comment]
Looks good, sans fixing the 2 issue (and tests!) :)

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala (hunk 1 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: +1. Thanks!

**Comment by eishay** [comment]
@andrewconner @raymondng with testing

**Comment by raymondng** [comment]
Nice :+1:

**Pull request merged**